2019-04-21 15:10:56,444 ERROR: Exception on /whisky_list [GET] [in /Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/masaki/whisky/app/main/routes.py", line 93, in whisky_list
    return render_template('whisky_list.html', title='All distilleries', all_distillery=all_distillery)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/templating.py", line 135, in render_template
    context, ctx.app)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/templating.py", line 117, in _render
    rv = template.render(context)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/masaki/whisky/app/templates/whisky_list.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/Users/masaki/whisky/app/templates/base.html", line 72, in top-level template code
    {% block app_content %}{% endblock app_content %}
  File "/Users/masaki/whisky/app/templates/whisky_list.html", line 12, in block "app_content"
    {% include "_distillery.html" %}
  File "/Users/masaki/whisky/app/templates/_distillery.html", line 8, in top-level template code
    <a href="{{ url_for('main.distillery', id=distillery.id) }}">{{ distillery.name }}</a>
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/helpers.py", line 356, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 2061, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/helpers.py", line 345, in url_for
    force_external=external)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/werkzeug/routing.py", line 2181, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'main.distillery' with values ['id']. Did you forget to specify values ['id_num']?
2019-04-21 15:12:14,933 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-21 15:13:15,922 ERROR: Exception on /whisky/1/submit [POST] [in /Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/masaki/whisky/app/main/routes.py", line 171, in submit_review
    review.add_tag(Tag.query.filter_by(name=tag).first())
  File "/Users/masaki/whisky/app/models.py", line 80, in add_tag
    if not self.is_tagged(tag):
  File "/Users/masaki/whisky/app/models.py", line 88, in is_tagged
    return self.tags.filter(Tag.id == tag.id).count() > 0
AttributeError: 'NoneType' object has no attribute 'id'
2019-04-21 15:16:37,664 ERROR: Exception on /whisky/1/submit [POST] [in /Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/masaki/whisky/app/main/routes.py", line 171, in submit_review
    review.add_tag(Tag.query.filter_by(name=tag).first())
  File "/Users/masaki/whisky/app/models.py", line 80, in add_tag
    if not self.is_tagged(tag):
  File "/Users/masaki/whisky/app/models.py", line 88, in is_tagged
    return self.tags.filter(Tag.id == tag.id).count() > 0
AttributeError: 'NoneType' object has no attribute 'id'
2019-04-21 15:25:17,646 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-21 15:26:51,210 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-21 15:27:11,474 ERROR: Exception on /whisky/1/submit [POST] [in /Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/masaki/whisky/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/masaki/whisky/app/main/routes.py", line 171, in submit_review
    review.add_tag(Tag.query.filter_by(name=tag).first())
  File "/Users/masaki/whisky/app/models.py", line 80, in add_tag
    if not self.is_tagged(tag):
  File "/Users/masaki/whisky/app/models.py", line 88, in is_tagged
    return self.tags.filter(Tag.id == tag.id).count() > 0
AttributeError: 'NoneType' object has no attribute 'id'
2019-04-22 11:12:33,126 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 11:15:33,302 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 11:17:40,926 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 11:26:38,015 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 11:29:12,942 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 11:29:45,405 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:14:52,249 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:36:46,955 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:39:56,474 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:40:08,724 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:40:35,865 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:43:05,483 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:51:36,873 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:52:26,400 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
2019-04-22 12:55:37,608 INFO: Whisky Blog startup [in /Users/masaki/whisky/app/__init__.py:71]
